src/dc_tools.rs

```diff
@@ -1069,137 +1084,88 @@ pub unsafe fn dc_get_abs_path(
     pathNfilename_abs
 }
 
-#[allow(non_snake_case)]
-pub unsafe fn dc_file_exist(context: &Context, pathNfilename: *const libc::c_char) -> libc::c_int {
-    let pathNfilename_abs = dc_get_abs_path(context, pathNfilename);
-    if pathNfilename_abs.is_null() {
-        return 0;
+pub fn dc_file_exist(context: &Context, path: *const libc::c_char) -> libc::c_int {
+    if let Some(path_abs) = dc_get_abs_path_safe(context, as_str(path)) {
+        std::path::Path::new(&path_abs).exists() as libc::c_int
+    } else {
+        0
     }
-
-    let exist = {
-        let p = std::path::Path::new(as_str(pathNfilename_abs));
-        p.exists()
-    };
-
-    free(pathNfilename_abs as *mut libc::c_void);
-    exist as libc::c_int
 }
 
-#[allow(non_snake_case)]
-pub unsafe fn dc_get_filebytes(context: &Context, pathNfilename: *const libc::c_char) -> uint64_t {
-    let pathNfilename_abs = dc_get_abs_path(context, pathNfilename);
-    if pathNfilename_abs.is_null() {
-        return 0;
-    }
-
-    let p = std::ffi::CStr::from_ptr(pathNfilename_abs)
-        .to_str()
-        .unwrap();
-    let filebytes = match fs::metadata(p) {
-        Ok(meta) => meta.len(),
-        Err(_err) => {
-            return 0;
+pub fn dc_get_filebytes(context: &Context, path: *const libc::c_char) -> uint64_t {
+    if let Some(path_abs) = dc_get_abs_path_safe(context, as_str(path)) {
+        match fs::metadata(&path_abs) {
+            Ok(meta) => meta.len() as uint64_t,
+            Err(_err) => 0,
         }
-    };
-
-    free(pathNfilename_abs as *mut libc::c_void);
-    filebytes as uint64_t
-}
-
-#[allow(non_snake_case)]
-pub unsafe fn dc_delete_file(context: &Context, pathNfilename: *const libc::c_char) -> libc::c_int {
-    let mut success: libc::c_int = 0i32;
-    let pathNfilename_abs = dc_get_abs_path(context, pathNfilename);
-    if pathNfilename_abs.is_null() {
-        return 0;
+    } else {
+        0
     }
-    let p = std::path::Path::new(
-        std::ffi::CStr::from_ptr(pathNfilename_abs)
-            .to_str()
-            .unwrap(),
-    );
+}
 
-    let res = if p.is_file() {
-        fs::remove_file(p)
-    } else {
-        fs::remove_dir_all(p)
-    };
+pub fn dc_delete_file(context: &Context, path: *const libc::c_char) -> libc::c_int {
```

Comment:
This should take path as `Path` so no conversions are needed.  Also the return value should be either `bool` or `Result<(), io::Error>` (or whatever error type is applicable), maybe depending on which is more suitable for most callers.  I guess it should be bool since the callers who care about Result can just call `fs::remove_file()` themselves.

Reply:
I am going to keep `dc_delete_file` prototype in this PR. I am changing only the implementations here, not interfaces.